### DISCUSSION TITLE
global set up for artillery fargate tests

### DISCUSSION THREAD

**Discussion by zoe-jobson-IL** [artilleryio/artillery#3417] [Artillery]
We have a set up where we run a number of aws fargate containers for each run of each test, Each fargate container supports around 200 user with playwright - so we tend to need a lot of pods - the main difficulty is that I would like to run a single once and for all pods set up for the whole test run ( basically using the suggestions [here ](https://github.com/artilleryio/artillery/tree/main/examples/using-data-from-redis) to avoid reusing users - which has been causing us some issues. I have some code that I can use to pull the users from our framework and spit them out into redis for popping during test runs, but obviously I only want to run the code to populate the redis lists once per test run. I would also like to be able to automatically trash the saved data after the test run, but that is less important right now. we generally run our larger multi fargate tests via gha so I could go down that route too.

**Comment by hassy** [comment]
hi @zoe-jobson-IL 👋 Redis is a great way to go if you have a Redis instance already. We have an example here: https://github.com/artilleryio/artillery/tree/main/examples/using-data-from-redis

You'll need to run the setup/teardown script outside of Artillery for now - we want to support a global setup/teardown hook natively, but it's not there yet
